go/build-and-test/action.yml

```diff
@@ -0,0 +1,150 @@
+name: "Go Build & Test"
+description: "Build and test go repo"
+
+inputs:
+  FLIPPCIRCLECIPULLER_REPO_TOKEN:
+    description: Flipp circleci repo token
+    required: true
+  BUF_BUILD_USER:
+    description: Buf CI user stored as secret
+    required: false
+  BUF_BUILD_API_TOKEN:
+    description: Buf API token stored as secret
+    required: false
+  SLACK_CHANNEL_ID:
+    description: The Slack channel ID(s) to send the data to
+    required: false
+  SLACK_BOT_TOKEN:
+    description: The Slack bot token to pass the data to
+    required: false
+  DB_HOST:
+    description: Database host
+    required: false
+  DB_USER:
+    description: Database user
+    required: false
+  DB_PASSWORD:
+    description: Database password
+    required: false
+  DB_PORT:
+    description: Database port
+    required: false
+  TEST_TAGS:
+    description: Test tags
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set branch variable
+      id: set-branch
+      shell: bash
+      run: |
+        if [ "${{ github.event_name }}" == "pull_request" ]; then
+          branch="${{ github.event.pull_request.head.ref }}"
+        else
+          branch="${{ github.ref }}"
+          branch="${branch#refs/heads/}"
+        fi
+        echo "branch=${branch}" >> $GITHUB_OUTPUT
+    - name: Get short sha
+      id: get-short-sha
+      run: echo "short_sha=`echo ${GITHUB_SHA::7}`" > $GITHUB_OUTPUT
+      shell: bash
+    - name: Configure Go environment
+      uses: wishabi/github-actions/go/configure@v0
+      with:
+        FLIPPCIRCLECIPULLER_REPO_TOKEN: ${{ inputs.FLIPPCIRCLECIPULLER_REPO_TOKEN }}
+        BUF_BUILD_USER: ${{ inputs.BUF_BUILD_USER }}
+        BUF_BUILD_API_TOKEN: ${{ inputs.BUF_BUILD_API_TOKEN }}
+    - name: Generate local protos
+      run: make generate-buf
+      shell: bash
+    - name: Setup Go dependencies
+      uses: wishabi/github-actions/go/deps@v0
+      with:
+        FLIPPCIRCLECIPULLER_REPO_TOKEN: ${{ inputs.FLIPPCIRCLECIPULLER_REPO_TOKEN }}
+    - name: Install mockery
+      run: go install github.com/vektra/mockery/v2
+      shell: bash
+    - name: Generate mocks
+      run: make generate-mocks
+      shell: bash
```

Comment:
Same here, not everyone would need mockery.
Ideally, we make these optional/conditional or add to the actions description, that repos using them should have a makefile with mockery commands inside

Reply:
Is there an easy way to detect if a repo depends on Mockery? Then I can just add a conditional and call `mockery` explicitly.

.... or we just do the PDA folder idea like you said and limit the scope to our team.

Reply:
Tried this:

```yaml
- name: Generate mocks
      if: ${{ hashFiles('.mockery.yml') != '' }}
      run: mockery
```

It worked: https://github.com/wishabi/distribution-api/actions/runs/6971579247/job/18981313019

Good enough?

Reply:
Neat! We could do the same with buf above using `buf.gen.yaml`

Reply:
Good call. Done.